ocaml/src/fragment_helper.ml

```diff
@@ -104,13 +119,46 @@ let maybe_decrypt
         Cipher.with_t_lwt
           key Cipher.AES256 Cipher.CBC []
           (fun cipher ->
-           Cipher.decrypt
-             ~iv
-             cipher
-             data 0 (Lwt_bytes.length data)) >>= fun () ->
+            Cipher.set_iv cipher iv;
+            Cipher.decrypt_detached
+                  cipher
+                  data 0 (Lwt_bytes.length data)
+          ) >>= fun () ->
         Lwt.return (Padding.unpad data)
+      | AES (CTR, L256) ->
+        Encryption.verify_key_length algo key;
+        get_iv algo key ~object_id ~chunk_id ~fragment_id ~ignore_fragment_id >>= fun iv ->
+        Cipher.with_t_lwt
+          key Cipher.AES256 Cipher.CTR []
+          (fun cipher ->
+            Cipher.set_ctr cipher iv;
+            Cipher.decrypt_detached
+                  cipher
+                  data 0 (Lwt_bytes.length data)
+          ) >>= fun () ->
+        Lwt.return (Bigstring_slice.wrap_bigstring data)
```

Comment:
Fwiw it's not only the last step (unpadding) that differs, also the chosen mode (CBC/CTR) and set_iv/set_ctr differ. Lifting the commonalities highlights the differences, so I'll do that